internal/provider/data_source_certificate_test.go

```diff
@@ -639,3 +639,45 @@ func TestDataSourceCertificate_MalformedURL(t *testing.T) {
 		},
 	})
 }
+
+func TestDataSourceCertificate_UnknownComputedCertificatesUntilApplied(t *testing.T) {
+	r.Test(t, r.TestCase{
+		ProtoV5ProviderFactories: protoV5ProviderFactories(),
+
+		Steps: []r.TestStep{
+			{
+
+				Config: `
+					resource "tls_private_key" "test" {
+						algorithm = "ED25519"
+					}
+
+					data "tls_certificate" "test" {
+						url = replace(tls_private_key.test.id, "/^.*$/","https://terraform.io")
+					}
+
+					output "test" {
+						value = data.tls_certificate.test.certificates[0].sha1_fingerprint
```

Comment:
Nit: Similarly here, for future readers

```suggestion
						# This test must reference an underlying value of the
						# certificates attribute to trigger the behavior.
						value = data.tls_certificate.test.certificates[0].sha1_fingerprint
```